h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java

```diff
@@ -74,6 +77,79 @@ public void initActualParamValues() {
     //the training frame used for blending (from which predictions columns are computed)
     public Key<Frame> _blending;
 
+    public enum MetalearnerTransform {
+      NONE,
+      Logit,
+      PercentileRank;
+
+      public Frame transform(StackedEnsembleModel model, Frame frame) {
+        if (this == Logit) {
+          return new MRTask() {
+            @Override
+            public void map(Chunk[] cs, NewChunk[] ncs) {
+              LinkFunction logitLink = LinkFunctionFactory.getLinkFunction(LinkFunctionType.logit);
+              for (int c = 0; c < cs.length; c++) {
+                for (int i = 0; i < cs[c]._len; i++) {
+                  final double p = Math.min(1 - 1e-9, Math.max(cs[c].atd(i), 1e-9)); // 0 and 1 don't work well with logit
+                  ncs[c].addNum(logitLink.link(p));
+                }
+              }
+            }
+          }.doAll(frame.numCols(), Vec.T_NUM, frame)
+                  .outputFrame(frame._key, frame._names, null);
+        } else if (this == PercentileRank) {
+          int N = 128;
+          // quantile indices, e.g., 0.5 is median etc
+          double[] probs = new double[N+1];
+          for (int i = 0; i <= N; i++)
+            probs[i] = i/(double)N;
+          if (null == model._output._metalearner_percentile_rank_precomputed_quantiles) {
+            // Compute quantiles for specified probs
+            DKV.put(frame);
+            QuantileModel.QuantileParameters qp = new QuantileModel.QuantileParameters();
+            qp._train = frame._key;
+            qp._probs = probs;
+            QuantileModel qm = new Quantile(qp).trainModel().get();
+            model.write_lock();
+            model._output._metalearner_percentile_rank_precomputed_quantiles = qm._output._quantiles;
```

Comment:
Is this transform only applied in training? Or in scoring a test set as well?

If it is not just training you would be overwriting the pre-determined quantiles.

Reply:
not true ^^ please disregard